File: Elixir/factorial.exs
defmodule Factorial do
    def f(1) do 1 end
    def f(n) do n * f(n - 1) end
end

IO.puts Factorial.f(1) # 1
IO.puts Factorial.f(5) # 120
IO.puts Factorial.f(7) # 5040

File: Go/factorial.go
package main

import "fmt"

func factorial(n int) int {
	result := 1
	for i := n; i != 0; i-- {
		result *= i
	}

	return result
}

func main() {
	fmt.Println(factorial(1)) // 1
	fmt.Println(factorial(5)) // 120
	fmt.Println(factorial(7)) // 5040
}